### PR TITLE
fix: properly stringify optional public message

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

The webui error handler accepts a public message to show for errors but when logging the message can sometimes show `undefiend` as part of the log statement.

Screen Shot 2020-05-27 at 2 32 22 PM

## Test Plan

Screen Capture_select-area_20200527131613
Screen Capture_select-area_20200527131521

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/utils/string.ts [hunk 1]
```diff
@@ -18,3 +18,8 @@ export const toHtmlId = (str: string): string => {
     .replace(/[^a-z0-9-]/gi, '')
     .toLowerCase();
 };
+
+// prepare an optional string to be used when concatnating strings.
+export const optionalStr = (str?: string, suffix='', prefix=''): string => {
```
Comment (stoksc) [inline webui/react/src/utils/string.ts:23]: seems a little overkill to me, but I'll trust your judgement more than mine on what functions would be useful in frontend world

Comment (hamidzr) [reply webui/react/src/utils/string.ts:23]: yeah, I'm torn here also. the idea is to save some characters or having to define a separate variable outside where you're building a string. especially when the inputs have long variable names :thinking: 
the performance hit of additional fn call wouldn't be concerning here.

Comment (hkang1) [reply webui/react/src/utils/string.ts:23]: suggestion:
```
export const listToStr = (list: string[], glue = ' '): string => {
  return list.filter(item => !!item).join(glue);
};
```

File: webui/react/src/ErrorHandler.ts [hunk 2]
```diff
@@ -83,7 +84,7 @@ const handleError = (e: DaError): Error => {
 
   // log the error if needed.
   if (!e.silent) {
-    const msg = `${e.type}: ${e.publicMessage + ' ' || ''}${e.message}`;
+    const msg = `${e.type}: ${optionalStr(e.publicMessage, ' ')}${e.message}`;
```
Comment (hkang1) [inline webui/react/src/ErrorHandler.ts:87]: based on suggestion below:
```
const msg = listToStr([ `${e.type}:`, e.publicMessage, e.message ]);
```

**Pull request merged**